lib/src/main/java/com/scalar/admin/kubernetes/Pauser.java

```diff
@@ -123,17 +149,33 @@ private void unpauseWithRetry(
         return;
       } catch (Exception e) {
         if (++retryCounter >= maxRetryCount) {
-          logger.warn(
+          // If someone uses this library directly instead of using our CLI, users should handle the
+          // exception properly. However, this case is a very critical issue. Therefore, we output
+          // the error message here despite whether the exception is handled or not on the caller
+          // side.
+          logger.error(
               "Failed to unpause Scalar product. They are still in paused. You must restart related"
                   + " pods by using the `kubectl rollout restart deployment {}`"
-                  + " command to unpase all pods.",
+                  + " command to unpause all pods.",
               target.getDeployment().getMetadata().getName());
-          return;
+          // In our CLI, we catch this exception and output the message as an error on the CLI side.
+          throw new PauserException(
+              String.format(
+                  "Failed to unpause Scalar product. They are still in paused. You must restart"
+                      + " related pods by using the `kubectl rollout restart deployment %s` command"
+                      + " to unpause all pods.",
+                  target.getDeployment().getMetadata().getName()),
+              e);
```

Comment:
If `unpause()` failed three times, throw the `PauserException`. The caller of the `pause()` method should handle this exception.

I will update the message after I implement the `pauser.unpause()` method. In other words, I will add a new option (or subcommand) that allows users to run the unpause operation solely, in another PR.

After I implement such a new feature, users can use it to unpause Scalar products instead of using the `kubectl rollout restart deployment` command.